src/Entity/Programme.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\Common\Collections\Collection;
+
+class Programme
+{
+    private int $id;
+
+    public string $name = '';
+
+    public string $description = '';
+
+    private \DateTime $startDate;
+
+    private \DateTime $endDate;
+
+    private ?User $trainer;
+
+    private Room $room;
+
+    private ?Collection $customers;
+
+    public bool $isOnline;
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getStartDate(): \DateTime
+    {
+        return $this->startDate;
+    }
+
+    public function setStartDate($startDate): self
+    {
+        return $this->startDate = $startDate;
```

Comment:
:D this will break because you have self as return type. as an exercise, please see what this method actually returns in this form.

Reply:
It expects the return value to be an instance of Programme, not DateTime